### PR TITLE
list structure

### PR DESCRIPTION
**Pull request by ReemMohamedAbdelfatah**
-Setting web pack configurations
-Setting linter configurations.
-Build the list structure.

### CODE REVIEW

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @ReemMohamedAbdelfatah  :wave: ,
I am Nemwel your reviewer for this PR :man_beard: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## :heavy_multiplication_x:  Invalid Code Review Request :heavy_multiplication_x: 

You have submitted a project, that does not have a professional README. Kindly update your README with the Microverse template and request for a review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by frank1738 — COMMENTED** [review]
Hi @ReemMohamedAbdelfatah , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has no professional readme. Kindly use the microverse template provided and customize it. Also, don't forget to include instructions on how a user can install the project and use it on their local machines

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/index.js [hunk 1]
```diff
@@ -1,14 +1,35 @@
...
+{
+  description: 'Buy groceries',
+  isCompleted: false,
+  index: 1,
+},
+{
+  description: 'Cook dinner',
+  isCompleted: false,
+  index: 2,
+}];
```
Comment (codecaiine) [inline src/index.js]: - [x] on load our list is great rendered to the DOM :clap:  But from the project requirement you need to render them in the order of their index. When I changed the index value the order doesn't change :cry:  Please fix it in order to make your project work as expected :smile: 

I think this piece of code can help you fix it :arrow_right:   **Array.sort((a, b) => a.index - b.index**

File: src/style.css [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+}
+
+i {
+  opacity: 0.4;
+  cursor: move;
+}
+
+section {
+  width: 50%;
+}
```
Comment (codecaiine) [inline src/style.css]: - [x] Please work on your design and make to make it similar to the project captured in the video below :arrow_down: Note that as a developer it is very important to pay attention to detail. :smile: 

#### Project capture
68747470733a2f2f696d672e796f75747562652e636f6d2f76692f416355642d5f596a6a71672f302e6a7067 (1)

#### Your todo list design
Screenshot from 2022-10-08 13-37-57

File: src/usingModules.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+const counter = -1;
+
+const AddToDo = (description) => {
+  const isCompleted = 'false';
+  const index = counter+1;
+  counter++;
+const newToDo = new todo(description, isCompleted, index);
+list.push(newToDo);
+}
+*/
```
Comment (codecaiine) [inline src/usingModules.js]: - [x] Removing commented code not only makes it harder to find later, but it also makes it so people in the future don't know it existed before. For finding the code, there are [tools and git commands](http://stackoverflow.com/q/278192/971592) to help you look at the history of a file. Please remove all commented code and unnecessary files from your code base :smile:

File: README.md [hunk 4]
```diff
@@ -1,2 +1,97 @@
...
+- It simply lists the things that you need to do and allows you to mark them as complete, and it uses ES6 and Webpack.
+- Single page website.
+- configure with linters.
+
+## Screenshot
+
+![readme](./assets/screenshoots/M1.png)
+![readme](./assets/screenshoots/M2.png)
+![readme](./assets/screenshoots/D1.png)
+![readme](./assets/screenshoots/D2.png)
```
Comment (codecaiine) [inline README.md]: - [x] Your screenshots links are broken , please fix it in order to make them appear in your readme and make it more professional :smile:

File: README.md [hunk 5]
```diff
@@ -1,2 +1,97 @@
...
+
+# Live Demo
+
+[Live Demo Link](https://reemmohamedabdelfatah.github.io/TO-DO-LIST/)
```
Comment (codecaiine) [inline README.md]: - [x] Your live demo link is broken :cry: Please fix it in order to make it available and your readme more professional :+1:

File: README.md [hunk 6]
```diff
@@ -1,2 +1,97 @@
...
+- GitHub: [@github](https://github.com/ReemMohamedAbdelfatah)
+- LinkedIn: [LinkedIn](https://www.linkedin.com/in/reem-abd-el-fatah-a07543116)
+
+## Acknowledgments
+
+- Hat tip to anyone whose code was used
+- Inspiration
+- etc
+- Design
+Credit goes to **Cindy Shin's** design on [Behance] (https://www.behance.net/gallery/29845175/CC-Global-Summit-2015). This is a mockup of Cindy's design.
```
Comment (codecaiine) [inline README.md]: - [x] I noticed that this information is not useful and doesn't have to be here please remove it and make sure to remove all unnecessary information from your readme to make it more professional  :+1:

**Review by codecaiine — CHANGES_REQUESTED** [review]
Hi @ReemMohamedAbdelfatah  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Great use of GitHub flow :heavy_check_mark: 
- No linters errors :heavy_check_mark: 
- Great project implementation :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/usingModules.js (hunk 3 above)
Comment (ReemMohamedAbdelfatah) [reply src/usingModules.js]: Thank you @codecaiine for sharing this link!

File: src/index.js (hunk 1 above)
Comment (ReemMohamedAbdelfatah) [reply src/index.js]: Thank you @codecaiine for your assistance.

**Review by frank1738 — APPROVED** [review]
Hi @ReemMohamedAbdelfatah , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

YesYouAreGIF (2)


## Highlights

✅ Correct use of Github workflow
✅ No linter error
✅ README is okay

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BenMukebo — APPROVED** [review]
### Hi @ReemMohamedAbdelfatah, 👋 

# STATUS: Approved  🎉 

Great work on making the changes requested by a previous reviewer :1st_place_medal:  
 Your project is complete!  There is nothing else to say other than... it's time to merge it
Congratulations! 🎉  🎉 🎉 



### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @BenMukebo ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**